### PR TITLE
Added Js Week2 Homework

### PR DESCRIPTION
**Pull request by semih1239**

### CODE REVIEW

File: javascript/javascript1/week2/script.js [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    return console.log('Lord ' + fullname1 + ' ' + fullname2)
+    }
+    else{return console.log(fullname1 + ' ' + fullname2)}
+}
+getFullname(name1,name2,useFormalName)
+
+
+// Event application
+
+let days = ['Sunday','Monday','Tuesday','Wednesday','Thursday','Friday','Saturday']
```
Comment (FrederikNygart) [inline javascript/javascript1/week2/script.js]: will this ever change? consider another declaration option.

File: javascript/javascript1/week2/script.js [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+function getEventWeekdays(number){
+    let today = new Date()
+    let day = today.getDate()
+    today.setDate(day+number)
+    return console.log(days[today.getDay()])
```
Comment (FrederikNygart) [inline javascript/javascript1/week2/script.js]: Interesting idea to just use the date api instead of remainder suggestion.

Unfortunately this will not work.

let's take today as an example(tuesday the 11th of may). Get date will return 11 as a result. Given 'number' is set to 2 we will increment the date by 13 days ending up with a **Monday** instead of the requested **Thursday** - _ as 2 days from now it will be thursday_.

also you should not return console.log() statements. console.log is used to console output(for debugging) while return statements are returning a value for you to use further on. 

You could perhaps instead change it to this:

```js
console.log(getEventWeekdays(2))
```

now we console log the **returned** result of the function.

File: javascript/javascript1/week2/script.js [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+}
+
+getEventWeekdays(55)
+
+
+// Weather wear
+
+function whatAmIWear(temp){
```
Comment (FrederikNygart) [inline javascript/javascript1/week2/script.js]: From the name 'What am i wear' it is difficult to understand the intention of the function. 

Your function should inform about 'what' it does. In this case i would say it: 'gets appropriate clothing by tempurature in celsius'.

This what you need to communicate to other people who potentially wants to use your code(and to yourself 3 months from now, when you have forgotten everything you ever knew about this code - trust me it will happen 🤕 )

TLDR;

maybe consider this name: getClothesToWear(temperatureInCelsius)

File: javascript/javascript1/week2/script.js [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+    if(temp>=25){
+        console.log('Shorts and T-shirt')
```
Comment (FrederikNygart) [inline javascript/javascript1/week2/script.js]: return the result instead of logging

File: javascript/javascript1/week2/script.js [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+        console.log('Coats and Boots')
+    }
+    else if(temp>-40){
+        console.log('Wear Everything Double')
+    }
+    else{
+        console.log("Dont't Go Outside")
+    }
+}
+whatAmIWear(-20)
```
Comment (FrederikNygart) [inline javascript/javascript1/week2/script.js]: console log the result

File: javascript/javascript1/week2/script.js [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+
+
+// Student manager
+
+let class18Students = []
+function addStudentToClass(studentName) {
+    if (studentName == 'Margrethe') {
```
Comment (FrederikNygart) [inline javascript/javascript1/week2/script.js]: couldn't many have the name Margrethe(maybe we need to be more specific)

File: javascript/javascript1/week2/script.js [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+
+// Formal fullname
+
+let name1 = prompt('Please write your name')
+let name2 = prompt('Please write your surname')
+useFormalName = confirm('Do you want formal name?')
+
+function getFullname(fullname1,fullname2,useFormalName){
+    if (useFormalName){
+    return console.log('Lord ' + fullname1 + ' ' + fullname2)
```
Comment (FrederikNygart) [inline javascript/javascript1/week2/script.js]: return value instead of console.log

File: javascript/javascript1/week2/script.js [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+    }
+    else{return console.log(fullname1 + ' ' + fullname2)}
+}
+getFullname(name1,name2,useFormalName)
```
Comment (FrederikNygart) [inline javascript/javascript1/week2/script.js]: log result

File: javascript/javascript1/week2/script.js [hunk 9]
```diff
@@ -0,0 +1,130 @@
+// Flight booking fullname function
+
+function getFullname(fullname1,fullname2){
+    return console.log(fullname1 + ' ' + fullname2)
```
Comment (FrederikNygart) [inline javascript/javascript1/week2/script.js]: return value instead of console.log

File: javascript/javascript1/week2/script.js [hunk 10]
```diff
@@ -0,0 +1,130 @@
...
+
+
+// Student manager
+
+let class18Students = []
+function addStudentToClass(studentName) {
+    if (studentName == 'Margrethe') {
+        class18Students.push(studentName)
+    }
+    else if (class18Students.indexOf(studentName) != (-1)) {
```
Comment (FrederikNygart) [inline javascript/javascript1/week2/script.js]: no need for parentheses around -1

File: javascript/javascript1/week2/script.js [hunk 11]
```diff
@@ -0,0 +1,130 @@
...
+        console.log(`Student ${studentName} is already in the class`)
+    }
+    else if (class18Students.length>=6){
+        console.log('Cannot add more students to class 18')
+    }
+    else if(studentName==''){
+        console.log('You cant add "nothing"')
+    }
+    else{
```
Comment (FrederikNygart) [inline javascript/javascript1/week2/script.js]: redundant else block

File: javascript/javascript1/week2/script.js [hunk 12]
```diff
@@ -0,0 +1,130 @@
...
+    if (studentName == 'Margrethe') {
+        class18Students.push(studentName)
+    }
+    else if (class18Students.indexOf(studentName) != (-1)) {
+        console.log(`Student ${studentName} is already in the class`)
+    }
+    else if (class18Students.length>=6){
+        console.log('Cannot add more students to class 18')
+    }
+    else if(studentName==''){
```
Comment (FrederikNygart) [inline javascript/javascript1/week2/script.js]: what if i call your function with the following values:
```js
addStudentToClass(null) // adds null
addStudentToClass() //adds undefined
addStudentToClass(' ') //adds empty string
```

and - bonus question - what if i call your function with a number or boolean value?

File: javascript/javascript1/week2/script.js [hunk 13]
```diff
@@ -0,0 +1,130 @@
...
+        total = total + boughtCandyPrices[i]
+    }
+    if(total < amountToSpend){
+        console.log(`You can buy ${amountToSpend-total} money`)
+    }
+    else{
+        console.log('You cant buy more candy.')
+    }
+}
+canBuyMoreCandy()
```
Comment (FrederikNygart) [inline javascript/javascript1/week2/script.js:137]: good job!

**Review by FrederikNygart — CHANGES_REQUESTED** [review]
Nice work Semih, I have a good amount of comments, but overall i would say you understand the primary concepts of all these exercises. 

Let me know if you need help with any of my comments.

**Comment by semih1239** [comment]
> Nice work Semih, I have a good amount of comments, but overall i would say you understand the primary concepts of all these exercises.
> 
> Let me know if you need help with any of my comments.

Hi . I updated all homeworks. Thank your very much for your feedbacks. 👍
